pkg/test/framework/components/istio/operator.go

```diff
@@ -260,14 +260,28 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 	if !i.isExternalControlPlane() {
```

Comment:
great find, would it make more sense to tell Istio that we're running in External Control Plane mode, which we kinda are,
and we'd end up skipping this block of code

Reply:
not possible, first i tried that only. external cluster decision is based on `controlCluster` and `!configCluster`. Ideally cluster is both the controlCluster and the configcluster, to match `!configCluster`, will have to add hack in in test setup to override kube setup. I actually tried that. But this hack bites back again at other place
```
  if !i.environment.IsConfigCluster(cluster) {                         var cluster resource.Cluster 
                configCluster, err := i.environment.GetConfigCluster(cluster)
                if err != nil {
                        return err
                }
                istiodAddress, err = i.RemoteDiscoveryAddressFor(configCluster)
                if err != nil {
                        return err
                }
                if err := patchIstiodCustomHost(istiodAddress, cfg, cluster); err != nil {
                        return err
                }
                if err := configureDiscoveryForConfigCluster(istiodAddress.IP.String(), cfg, configCluster); err != nil {
                        return err
                }
        }


```

Reply:
after trying all this, patch here seemed cleaner to me.

Reply:
cool, you can also delete this code and move the comments to the commit message
since it code doesn't exist upstream - https://github.com/istio/istio/blob/79bcecfd7f44fbb91a971133b1add7e7db4bff8f/pkg/test/framework/components/istio/operator.go#L342

Reply:
or cherrypick the right upstream patch

Reply:
they just moved it a few lines up https://github.com/istio/istio/blob/79bcecfd7f44fbb91a971133b1add7e7db4bff8f/pkg/test/framework/components/istio/operator.go#L332-L338 :D

Reply:
ah thanks for surfacing it, so looks these two snippets don't gel well with XCP
```
	if ctx.Clusters().IsMulticluster() {
		// For multicluster, configure direct access so each control plane can get endpoints from all
		// API servers.
		if err := i.configureDirectAPIServerAccess(ctx, cfg); err != nil {
			return nil, err
		}
	}
```

and

```
// IsMulticluster is a utility method that indicates whether there are multiple Clusters available.
func (c Clusters) IsMulticluster() bool {
	return len(c) > 1
}
```

do you think it makes more sense to redefine `IsMulticluster` and raise a PR upstream ?

ps. - my comments are non blocking